### DISCUSSION TITLE
zh/models/rtdetr

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#24446] [Docs]
# zh/models/rtdetr

探索百度 RT-DETR，这是一款基于 Vision Transformer 的实时目标检测器，提供高精度和可调节的推理速度。通过 Ultralytics 了解更多信息。

https://docs.ultralytics.com/zh/models/rtdetr

**Comment by wba-zzu** [comment]
想要将原版RT-DETR中的ResNet50和ResNet18以及ResNet101完全移植到Ultralytics，并且获得其预训练模型

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for sharing your interest in Ultralytics and the Chinese RT-DETR documentation 🚀 This is an automated response to help get things started, and an Ultralytics engineer will also assist here soon.

For anyone exploring RT-DETR, we recommend visiting the main [Docs](https://docs.ultralytics.com/) as well as the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage pages, where many common workflows and examples are already covered. The Chinese RT-DETR page you referenced is a great place to start for model-specific details.

If this discussion is related to a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments, with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by raimbekovm** [comment]
hey @wba-zzu, thanks for raising this.

quick rundown of where things stand:

**resnet50 / resnet101** - the architectures are already shipped as yaml configs:

- [`rtdetr-resnet50.yaml`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/rt-detr/rtdetr-resnet50.yaml)
- [`rtdetr-resnet101.yaml`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/rt-detr/rtdetr-resnet101.yaml)

you can build and train them straight from yaml:

```python
from ultralytics import RTDETR

model = RTDETR("rtdetr-resnet50.yaml")
model.train(data="coco.yaml", epochs=100, imgsz=640)
```

however, ultralytics does not publish pretrained weights for these. the only official rt-detr checkpoints in the [ultralytics/assets releases](https://github.com/ultralytics/assets/releases) are `rtdetr-l.pt` and `rtdetr-x.pt`, both of which use the hgnetv2 backbone (not resnet). this is also flagged in the [model docs](https://docs.ultralytics.com/models/rtdetr/#supported-tasks-and-modes) under "architecture-only variants".

**resnet18** - not shipped. there is no `rtdetr-resnet18.yaml` in the repo, so a resnet18 variant would need its own yaml and corresponding modules added.

**loading external pretrained weights**: ultralytics builds the model from yaml, so the state_dict keys follow a pattern like `model.<index>.layer.<...>.conv.weight`. weights from another implementation will not load directly - they would need a key-mapping / conversion step to match this layout.

hope this helps!
